tests/Unit/Evolution/DgSubcell/Test_PerssonTci.cpp

```diff
@@ -0,0 +1,146 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include <array>
+#include <cstddef>
+
+#include "DataStructures/DataBox/Tag.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Variables.hpp"
+#include "Domain/LogicalCoordinates.hpp"
+#include "Evolution/DgSubcell/PerssonTci.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/MakeArray.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace {
+namespace Tags {
+struct Scalar : db::SimpleTag {
+  using type = ::Scalar<DataVector>;
+};
+
+template <size_t Dim>
+struct Vector : db::SimpleTag {
+  using type = tnsr::I<DataVector, Dim>;
+};
+}  // namespace Tags
+
+template <size_t Dim>
+DataVector soln(const tnsr::I<DataVector, Dim, Frame::Logical>& coords,
+                const size_t number_of_modes_per_dim,
+                const std::array<double, Dim>& highest_coeffs) noexcept {
+  DataVector result =
+      Spectral::compute_basis_function_value<Spectral::Basis::Legendre>(
+          1, get<0>(coords));
+  for (size_t d = 1; d < Dim; ++d) {
+    result += Spectral::compute_basis_function_value<Spectral::Basis::Legendre>(
+        1, coords.get(d));
+  }
+
+  for (size_t d = 0; d < Dim; ++d) {
+    result += gsl::at(highest_coeffs, d) *
+              Spectral::compute_basis_function_value<Spectral::Basis::Legendre>(
+                  number_of_modes_per_dim - 1, coords.get(d));
+  }
+  return result;
+}
+
+template <size_t Dim, typename TagToCheck>
+void test_persson_impl(const size_t num_pts_1d,
+                       const std::array<double, Dim>& highest_coeffs,
+                       const size_t tensor_to_modify,
+                       const double persson_exponent,
+                       const bool expected_tci_triggered) noexcept {
+  CAPTURE(Dim);
+  CAPTURE(db::tag_name<TagToCheck>());
+  CAPTURE(num_pts_1d);
+  CAPTURE(highest_coeffs);
+  CAPTURE(tensor_to_modify);
+  CAPTURE(persson_exponent);
+  CAPTURE(expected_tci_triggered);
+  const double zero_cutoff = 1.0e-18;
+  const Mesh<Dim> dg_mesh{num_pts_1d, Spectral::Basis::Legendre,
+                          Spectral::Quadrature::GaussLobatto};
+  const auto logical_coords = logical_coordinates(dg_mesh);
+  const std::array<double, Dim> zero_coeffs = make_array<Dim>(0.0);
+
+  Variables<tmpl::list<Tags::Scalar, Tags::Vector<Dim>>> vars(
+      dg_mesh.number_of_grid_points());
+  if (tensor_to_modify == 0) {
+    get(get<Tags::Scalar>(vars)) =
+        soln(logical_coords, dg_mesh.extents(0), highest_coeffs);
+  } else {
+    get(get<Tags::Scalar>(vars)) =
+        soln(logical_coords, dg_mesh.extents(0), zero_coeffs);
```

Comment:
[optional] I found `highest_coeffs` vs `zero_coeffs` slightly confusing because it made me think you were distinguishing between the highest (most oscillatory) and lowest (constant term) coeffs... but I think you're distinguishing between if the highest coeff is zero or non-zero. What would you think of renaming to the pair (`oscillatory_coeffs` and `zero_coeffs`) to emphasize the content, or perhaps the pair (`oscillatory_highest_coeffs` and `zero_highest_coeffs`) to emphasize the content + the info about which coeff is described?

(if you make a change here, make it also in the other function below)